internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicMetadataFetcher.java

```diff
@@ -0,0 +1,673 @@
+
+  /**
+   * Blocking queue is used to ensure single-threaded access to a consumer as PubSubConsumerAdapter
+   * implementations are not guaranteed to be thread-safe.
+   */
+  private final BlockingQueue<PubSubConsumerAdapter> pubSubConsumerPool;
+  private final List<Closeable> closeables = new ArrayList<>(2);
+  private final ThreadPoolExecutor threadPoolExecutor;
+  private final PubSubAdminAdapter pubSubAdminAdapter;
+  private final TopicManagerStats topicManagerStats;
+
+  /**
+   * The following caches store metadata related to topics, including details such as the latest offset,
+   * earliest offset, and topic existence. Cached values are set to expire after a specified time-to-live
+   * duration. When a value is not present in the cache, it is synchronously. If a cached value has expired,
+   * it is asynchronously retrieved, and the cache. value is updated upon completion of the asynchronous fetch.
+   * To avoid overwhelming metadata fetcher, only one asynchronous request is issued at a time for a given
+   * key within the specific cache.
+   */
+  private final Map<PubSubTopic, ValueAndExpiryTime<Boolean>> topicExistenceCache = new VeniceConcurrentHashMap<>();
+  private final Map<PubSubTopicPartition, ValueAndExpiryTime<Long>> latestOffsetCache = new VeniceConcurrentHashMap<>();
+  private final Map<PubSubTopicPartition, ValueAndExpiryTime<Long>> earliestOffsetCache =
+      new VeniceConcurrentHashMap<>();
+  private final Map<PubSubTopicPartition, ValueAndExpiryTime<Long>> lastProducerTimestampCache =
+      new VeniceConcurrentHashMap<>();
+  private final long cachedEntryTtlInNs;
+
+  public TopicMetadataFetcher(
+      String pubSubClusterAddress,
+      TopicManagerContext topicManagerContext,
+      TopicManagerStats topicManagerStats,
+      PubSubAdminAdapter pubSubAdminAdapter) {
+    this.topicManagerStats = topicManagerStats;
+    this.pubSubAdminAdapter = pubSubAdminAdapter;
+    this.pubSubConsumerPool = new LinkedBlockingQueue<>(topicManagerContext.getTopicMetadataFetcherConsumerPoolSize());
+    this.cachedEntryTtlInNs = MILLISECONDS.toNanos(topicManagerContext.getTopicOffsetCheckIntervalMs());
+    PubSubMessageDeserializer pubSubMessageDeserializer = PubSubMessageDeserializer.getInstance();
+    for (int i = 0; i < topicManagerContext.getTopicMetadataFetcherConsumerPoolSize(); i++) {
+      PubSubConsumerAdapter pubSubConsumerAdapter = topicManagerContext.getPubSubConsumerAdapterFactory()
+          .create(
+              topicManagerContext.getPubSubProperties(pubSubClusterAddress),
+              false,
+              pubSubMessageDeserializer,
+              pubSubClusterAddress);
+
+      closeables.add(pubSubConsumerAdapter);
+      if (!pubSubConsumerPool.offer(pubSubConsumerAdapter)) {
+        throw new VeniceException("Failed to initialize topicMetadataFetcherPool");
+      }
+    }
+
+    threadPoolExecutor = new ThreadPoolExecutor(
```

Comment:
Can we add some comments here to give a note to the reader that all async method in this class are executed by this dedicated thread pool?